webapp/packages/plugin-datasource-transaction-manager/src/TransactionManagerService.ts

```diff
@@ -0,0 +1,143 @@
+ * you may not use this file except in compliance with the License.
+ */
+import { computed, makeAutoObservable } from 'mobx';
+
+import { ConnectionExecutionContextResource, ConnectionInfoResource, createConnectionParam } from '@cloudbeaver/core-connections';
+import { injectable } from '@cloudbeaver/core-di';
+import { NotificationService } from '@cloudbeaver/core-events';
+import { TaskScheduler } from '@cloudbeaver/core-executor';
+import { AsyncTaskInfoService, GraphQLService } from '@cloudbeaver/core-sdk';
+import { MetadataMap } from '@cloudbeaver/core-utils';
+import { ConnectionSchemaManagerService } from '@cloudbeaver/plugin-datasource-context-switch';
+
+import { TransactionExecutionContext } from './TransactionExecutionContext';
+
+const DEFAULT_AUTO_COMMIT = true;
+
+@injectable()
+export class TransactionManagerService {
+  get currentContext() {
+    return this.connectionSchemaManagerService.activeExecutionContext;
+  }
+
+  get autoCommitMode() {
+    const context = this.currentContext;
+
+    if (context) {
+      return context.autoCommit ?? DEFAULT_AUTO_COMMIT;
+    }
+
+    return DEFAULT_AUTO_COMMIT;
+  }
+
+  readonly transactions: MetadataMap<string, TransactionExecutionContext>;
+  readonly scheduler: TaskScheduler<string>;
+
+  constructor(
+    private readonly notificationService: NotificationService,
+    private readonly graphQLService: GraphQLService,
+    private readonly asyncTaskInfoService: AsyncTaskInfoService,
+    private readonly connectionExecutionContextResource: ConnectionExecutionContextResource,
+    private readonly connectionSchemaManagerService: ConnectionSchemaManagerService,
+    private readonly connectionInfoResource: ConnectionInfoResource,
+  ) {
+    this.scheduler = new TaskScheduler((a, b) => a === b);
+    this.transactions = new MetadataMap(
+      contextId => new TransactionExecutionContext(this.scheduler, this.connectionExecutionContextResource, contextId),
+    );
+
+    makeAutoObservable(this, {
+      currentContext: computed,
+      autoCommitMode: computed,
+    });
+  }
+
+  async setAutoCommit(connectionId: string, contextId: string, autoCommit: boolean) {
+    const transaction = this.transactions.get(contextId);
+
+    const task = this.asyncTaskInfoService.create(async () => {
+      const { taskInfo } = await this.graphQLService.sdk.asyncSqlSetAutoCommit({
+        connectionId,
+        contextId,
+        autoCommit,
+      });
+
+      return taskInfo;
+    });
+
+    try {
+      await transaction.run(
+        async () => await this.asyncTaskInfoService.run(task),
+        () => this.asyncTaskInfoService.cancel(task.id),
+        () => this.asyncTaskInfoService.remove(task.id),
+      );
+
+      this.connectionExecutionContextResource.markOutdated();
+    } catch (exception: any) {
+      this.notificationService.logException(exception, 'plugin_datasource_transaction_manager_commit_mode_fail');
+    }
+  }
+
+  async commit(connectionId: string, contextId: string) {
```

Comment:
if there is no connection we should not allow to commit the transaction. we can `logError` in this case

also an error if `transaction.context` is `undefined`

Reply:
Context cant exists without connection and if the context is undefined, the error will be thrown in by .run function